src/test/java/uk/gov/hmcts/reform/translate/service/DictionaryServiceTest.java

```diff
@@ -218,12 +241,180 @@ void shouldUpdateADictionaryForUserWithoutManageTranslationsRole() {
+            lenient().when(securityUtils.hasRole(any())).thenReturn(true);
+            given(applicationParams.getPutDictionaryS2sServicesBypassRoleAuthCheck())
+                .willReturn(Arrays.asList(DEFINITION_STORE));
+
+            given(securityUtils.getServiceNameFromS2SToken(CLIENTS2S_TOKEN)).willReturn(XUI);
+            given(securityUtils.hasAnyOfTheseRoles(anyList())).willReturn(false);
+
+            RequestErrorException roleMissingException = assertThrows(
+                RequestErrorException.class, () -> dictionaryService.putDictionaryRoleCheck(CLIENTS2S_TOKEN)
+            );
+            assertThat(roleMissingException).isInstanceOf(RequestErrorException.class);
+            assertEquals(
+                String.format(
+                    RequestErrorException.ERROR_MESSAGE, MANAGE_TRANSLATIONS_ROLE + "," + LOAD_TRANSLATIONS_ROLE
+                ),
+                roleMissingException.getMessage()
+            );
+        }
+
+
+        // Incorrect pay_load
+        @Test
+        void shouldFailUpdateADictionaryForUserWithoutManageTranslationsRoleAndNullPayload() {
+
+
+            final Dictionary dictionaryRequest = new Dictionary(null);
+            given(applicationParams.getPutDictionaryS2sServicesBypassRoleAuthCheck())
+                .willReturn(Arrays.asList(DEFINITION_STORE));
+            given(securityUtils.getServiceNameFromS2SToken(CLIENTS2S_TOKEN)).willReturn(DEFINITION_STORE);
+            given(securityUtils.hasRole(anyString())).willReturn(false);
+            dictionaryService.putDictionaryRoleCheck(CLIENTS2S_TOKEN);
+
+            BadRequestException badRequestException = assertThrows(
+                BadRequestException.class, () -> dictionaryService.putDictionary(dictionaryRequest)
+            );
+            assertThat(badRequestException).isInstanceOf(BadRequestException.class);
+            assertEquals(INVALID_PAYLOAD_FORMAT, badRequestException.getMessage());
+
+        }
+
+        @Test
+        void shouldFailUpdateADictionaryForUserWithoutManageTranslationsRoleAndIncorrectPayload() {
+
+
+            final Dictionary dictionaryRequest = new Dictionary(new HashMap<>());
+            given(applicationParams.getPutDictionaryS2sServicesBypassRoleAuthCheck())
+                .willReturn(Arrays.asList(DEFINITION_STORE));
+            given(securityUtils.getServiceNameFromS2SToken(CLIENTS2S_TOKEN)).willReturn(DEFINITION_STORE);
+            given(securityUtils.hasRole(anyString())).willReturn(false);
+            dictionaryService.putDictionaryRoleCheck(CLIENTS2S_TOKEN);
+
+            BadRequestException badRequestException = assertThrows(
+                BadRequestException.class, () -> dictionaryService.putDictionary(dictionaryRequest)
+            );
+            assertThat(badRequestException).isInstanceOf(BadRequestException.class);
+            assertEquals(INVALID_PAYLOAD_FORMAT, badRequestException.getMessage());
+
+        }
```

Comment:
Can we drop `dictionaryService.putDictionaryRoleCheck` from these tests so we are just testing the negative calls to `dictionaryService.putDictionary`.

Reply:
DONE

Reply:
Not done but happy to ignor to progress PR.